pkg/machines/components/vcpuModal.jsx

```diff
@@ -0,0 +1,192 @@
+            state.cores = divs[divs.length - 1];
+        }
+        state.threads = (this.state.max/(state.sockets*state.cores));
+        this.setState(state);
+    }
+
+    onThreadsChange (value) {
+        const state = { sockets: this.state.sockets, threads: this.state.threads };
+        state.threads = parseInt(value);
+        let divs = dividers(this.state.max / state.threads);
+        if (divs.indexOf(state.sockets) === -1) {
+            state.sockets = divs[divs.length - 1];
+        }
+        state.cores = (this.state.max/(state.sockets*state.threads));
+
+        this.setState(state);
+    }
+
+    onCoresChange (value) {
+        const state = { sockets: this.state.sockets, threads: this.state.threads };
+        state.cores = parseInt(value);
+        let divs = dividers(this.state.max / state.cores);
+        if (divs.indexOf(state.sockets) === -1) {
+            state.sockets = divs[divs.length - 1];
+        }
+        state.threads = (this.state.max/(state.sockets*state.cores));
+        this.setState(state);
+    }
+
+
+    render () {
+        return (<div className="modal-body">
+            <table className="vcpu-detail-modal-table">
+                <tr>
+                    <td>
+                        <table className='form-table-ct'>
+                            <InfoRecord
+                                descr={_("vCPU Count")}
+                                tooltip={_("Fewer than the maximum number of virtual CPUs should be enabled.")}
+                                value={<input id="machines-vcpu-count-field" type="number" className="form-control" value={this.state.count} onChange={this.onCountSelect} />}
+                            />
+                            <InfoRecord
+                                descr={_("vCPU Maximum")}
+                                tooltip={cockpit.format(_("Maximum number of virtual CPUs allocated for the guest OS, which must be between 1 and $0"), this.props.hypervisorMax)}
+                                value={<input id="machines-vcpu-max-field" type="number" className="form-control" onChange={this.onMaxChange} value={this.state.max} />}
+                            />
+                        </table>
+                    </td>
+                    <td>
+                        <table className='form-table-ct vcpu-detail-modal-right'>
+                            <InfoRecord descr={_("Sockets")} tooltip={_("Total number of CPU sockets.")} value={
+                                <Select id='socketsSelect' value={this.state.sockets.toString()} onChange={this.onSocketChange} items={dividers(this.state.max).map((t) => t.toString())} />
+                            }/>
+                            <InfoRecord descr={_("Cores per socket")} value={
+                                <Select id='coresSelect' value={this.state.cores.toString()} onChange={this.onCoresChange}  items={dividers(this.state.max).map((t) => t.toString())} />
+                            }/>
```

Comment:
This is dependent on number of CPUs and sockets, so I assume this is where all the divider logic comes into place. Why would this not just be a read-only value, or even not shown at all? If it makes sense to provide configurability of sockets, perhaps it could even be the other way around? Then the user can select arbitrary number of sockets and CPUs per socket, which then determines the total number of CPUs; this can then be shown as a read-only value, and there is less confusion about how to treat invalid values.

Reply:
It depends on maximum number of CPUs (Max), Sockets and Threads. Max = Sockets * Cores * Threads.